Library/Homebrew/utils/analytics.sh

```diff
@@ -0,0 +1,71 @@
+setup-analytics() {
+  if [[ -z "$HOMEBREW_ANALYTICS" || -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    return
+  fi
+
+  # User UUID file. Used for Homebrew user counting. Can be deleted and
+  # recreated with no adverse effect (beyond our user counts being inflated).
+  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  if [[ -r "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
+  then
+    HOMEBREW_ANALYTICS_USER_UUID="$(<"$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
+  else
+    HOMEBREW_ANALYTICS_USER_UUID="$(uuidgen)"
+    echo "$HOMEBREW_ANALYTICS_USER_UUID" > "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
+  fi
+  export HOMEBREW_ANALYTICS_ID="UA-75654628-1"
+  export HOMEBREW_ANALYTICS_USER_UUID
+}
+
+report-analytics-screenview-command() {
+  if [[ -z "$HOMEBREW_ANALYTICS" || -n "$HOMEBREW_NO_ANALYTICS" ]]
+  then
+    return
+  fi
+
+  # Don't report non-official commands.
+  if ! [[ "$HOMEBREW_COMMAND" = "bundle"   ||
+          "$HOMEBREW_COMMAND" = "cask"     ||
+          "$HOMEBREW_COMMAND" = "services" ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.rb"     ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh"     ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.rb" ||
+          -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]]
+  then
+    return
+  fi
+
+  # Don't report commands used mostly by our scripts and not users.
+  if [[ "$HOMEBREW_COMMAND" = "commands" ]]
```

Comment:
Clearly, current blacklist approach doesn't scale well. It may be better to put all autocompletion script related commands into a separate group at first. e.g. a dedicate command for script autocompletion? (just a random idea)